Produlytics/ui/back-end/src/main/java/it/deltax/produlytics/uibackend/security/LoginController.java

```diff
@@ -13,14 +17,19 @@
 public class LoginController {
   /** Riceve le chiamate all'endpoint REST per l'autenticazione. */
   @GetMapping
-  public Map<String, String> login(HttpSession session, Authentication authentication) {
+  public Map<String, Object> login(HttpSession session, Authentication authentication) throws BusinessException {
     if(authentication != null) {
       var authorities = authentication.getAuthorities();
       var adminAuthority = ProdulyticsGrantedAuthority.ADMIN.getAuthority();
       var admin = authorities.stream().anyMatch(auth -> auth.getAuthority().equals(adminAuthority));
-      return Map.of("accessToken", session.getId(), "admin", Boolean.toString(admin));
+      var username = authentication.getName();
+      return Map.of(
+          "accessToken", session.getId(),
+          "admin", admin,
+          "username", username
+          );
     } else {
-      return Map.of();
+      throw new BusinessException("unauthorized", ErrorType.UNAUTHORIZED);
```

Comment:
Lo consideriamo un errore di business? Si potrebbe ritornare direttamente 401...

Reply:
io lo considererei di business perché in teoria è una funzionalità del sistema

poi beh se facciamo la query a maggior ragione

Reply:
Se diventa un endpoint a parte allora puoi semplicemente modificare la configurazione di Spring Security per richiedere un utente autenticato.

Questo secondo me è un altro punto a favore nel fare un endpoint separato:
- login deve essere disponibile a tutti perchè nel momento in cui stai facendo il login non sei ancora autenticato;
- ottenere i dati della sessione attuale ha invece senso solo se esiste una sessione, e quindi sei già loggato.

Quell'`if` strano dentro a `login` e il bisogno di un errore di business 401 non sono altro che sintomi di questo problema.

Reply:
Meglio non ritornare la `BusinessException` in un controller. Era stata pensata per essere usata nei service -> business logic, quindi altra storia rispetto a un controller. Se quella logica la riteniamo logica di business spostiamola in un service dedicato, altrimenti concordo con il ritornare direttamente 401.